rita_common/src/rita_loop/write_to_disk.rs

```diff
@@ -0,0 +1,121 @@
+use crate::{
+    debt_keeper::send_debt_update, traffic_watcher::watch, tunnel_manager::tm_get_neighbors,
+};
+use std::{
+    thread,
+    time::{Duration, Instant},
+};
+
+use actix_async::System as AsyncSystem;
+use babel_monitor::{open_babel_stream, parse_routes};
+use settings::{
+    client::RitaClientSettings, exit::RitaExitSettingsStruct, get_rita_client, get_rita_exit,
+    FileWrite,
+};
+
+pub const FAST_LOOP_TIMEOUT: Duration = Duration::from_secs(4);
+// Save duration for all writes to disk in order to reduce write operations
+// pub const SAVING_TO_DISK_FREQUENCY: Duration = Duration::from_secs(600);
+#[derive(Clone)]
+pub enum OldSettingsType {
```

Comment:
In this case you're using 'old' to mean 'the settings on the disk' this took a couple of minutes to figure out. So the name/comment needs some improvement to indicate that maybe SettingsOnDisk?

Also I don't think we should pass the settings on the disk as an argument, nor the save interval. Only the file path since it's an argument.

Right now this loop never updates the old settings so once it' saves once it will keep saving again every loop iteration forever, you should declare a variable with the config on the disk at the top of the loop and update it every time you save. Save interval should come from the config.

Reply:
The issue was that calling this loop function needs the ability to discriminate between if the device was a client or an exit. I thought passing in the enum was the best way since if you just pass the file path I wasn't sure how to tell if the device was a client or exit. I think if you call settings::get_rita_common it isn't exactly clear how to tell what it is.

Reply:
I think in this case the best option may be to add a function in settings that returns an enum. I agree the enum is needed, but I don't think it needs to be an argument to the saving loop.